src/frontend/context/SecurityContext.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { ASYNC_KILL_KEY, PASSWORD_KEY } from "../constants";
+import { useEffect } from "react";
+import { useRef } from "react";
+
+//We should be setting the password to null when the password is turned off
+interface SecuritContextType {
+  killState: boolean;
+  setKillState: (killStateValue: boolean) => void;
+  passcode: string | null;
+  setPasscode: (passValue: string | null) => void;
+  checkFlag: boolean;
+  killStateActive: boolean;
+  setKillStateActive: (isActive: boolean) => void;
+  setCheckFlag: (checkFlag: boolean) => void;
+}
+
+const DEFAULT_CONTEXT: SecuritContextType = {
+  killState: false,
+  setKillState: () => {},
+  setPasscode: () => {},
+  passcode: null,
+  checkFlag: false,
+  setCheckFlag: () => {},
+  killStateActive: false,
+  setKillStateActive: () => {},
+};
+
+export const SecurityContext = React.createContext<SecuritContextType>(
+  DEFAULT_CONTEXT
+);
+
+export const SecurityProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [killState, setKillState] = React.useState(false);
+  const [killStateActive, setKillStateActive] = React.useState(false);
+  const [passcode, setPasscode] = React.useState<string | null>(null);
+  const [checkFlag, setCheckFlag] = React.useState(false);
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    if (firstRender.current) {
```

Comment:
is this check needed if this is an effect with a array dep of `[]`?

Reply:
So when a user open the app, the password must be set to the variable in memory (`passcode`). When `passcode` gets set, I have THIS listener which sets asyncstorage to the new `passcode`. This check basically just avoid the asyncstorage being unecessarily set the first time

Reply:
think this becomes less of a problem if you move the setItem calls out of the effect below (see [comment](https://github.com/digidem/mapeo-mobile/pull/923/files#r807390019))